java/code/src/com/redhat/rhn/common/hibernate/HibernateFactory.java

```diff
@@ -780,11 +781,33 @@ protected static void executeCallableMode(String name, String mode, Map params)
         return findByIds(ids, queryName, idsParameterName, new HashMap<String, Object>());
     }
 
+    /**
+     * Executes an 'update' query to the database given a list of parameters.
+     * The query will be execute in batches of LIST_BATCH_MAX_SIZE parameters each.
+     * @param <E> the type of the list parameters
+     * @param list the list of parameters to search for
+     * @param queryName the name of the query to be executed
+     * @param parameterName the name of the parameter to match the parameters in the list
+     * @return the count of affected rows
+     */
+    @SuppressWarnings("unchecked")
+    protected static <E> int udpateByIds(List<E> list, String queryName, String parameterName,
```

Comment:
Is there any particular deep reason why making this method `protected`?

Reply:
it's mean to be used only by subclasses

Reply:
Maybe I am not seeing something obvious. What would break if this was `public`?

Reply:
It wouldn't break anything, but it shouldn't be public. This method is quite invasive in the sense that it accepts a "query name", and that information should (in my opinion) be available only in those objects which responsibility is to access the DB. Such objects in the current code are sub-classes of "HibernateFactory", thus this method is protected and it's available only in the sub-classes.